pkg/security/admission/admission.go

```diff
@@ -95,9 +97,42 @@ func (c *constraint) Admit(a kadmission.Attributes) error {
 
 	// remove duplicate constraints and sort
 	matchedConstraints = oscc.DeduplicateSecurityContextConstraints(matchedConstraints)
-	sort.Sort(oscc.ByPriority(matchedConstraints))
 
-	providers, errs := oscc.CreateProvidersFromConstraints(a.GetNamespace(), matchedConstraints, c.client)
+	if a.GetOperation() == kadmission.Update {
+		for _, constraint := range  matchedConstraints {
+			if NewSccChecker(constraint).allowsPod(pod) {
+				// FIXME: do we need to assign annotation?
+				if pod.ObjectMeta.Annotations == nil {
+					pod.ObjectMeta.Annotations = map[string]string{}
+				}
+				pod.ObjectMeta.Annotations[allocator.ValidatedSCCAnnotation] = constraint.Name
+
+				// if any allow the pod as-is, succeed
+				return nil
+			}
+		}
+
+		// otherwise, reject
+		return kadmission.NewForbidden(a, errors.New("TODO: request rejected because all SCCs forbid this pod"))
+	}
+
+	// 2. filter to ones that do not forbid the pod - TODO
+	var filteredConstraints []*kapi.SecurityContextConstraints
+	for _, constraint := range  matchedConstraints {
+		if NewSccChecker(constraint).allowsPod(pod) {
```

Comment:
In other words, "could allow pod" check is weaker than "allows pod as-is" because the former suppose that on step later we'll generate default values for a pod spec. Is it correct?